relay-metrics/src/aggregator.rs

```diff
@@ -95,7 +95,7 @@ impl BucketKey {
 /// Note that this does not necessarily match the exact memory footprint of the tags,
 /// because data structures or their serialization have overheads.
 pub fn tags_cost(tags: &BTreeMap<String, String>) -> usize {
-    tags.iter().map(|(k, v)| k.capacity() + v.capacity()).sum()
+    tags.iter().map(|(k, v)| k.len() + v.len()).sum()
```

Comment:
This is an unrelated bugfix that showed up in one of the tests: By using capacity, we were overestimating the size of the buckets. Due to the introduction of `unescaper::unescape`, the allocated tag strings were larger than their contents, which changed test behavior.

Reply:
It was deliberate, I originally used `capacity()` since we used this to control memory consumption. More than that, we now rely on this for serialization now and Joris has since changed the corresponding check on metric names to use `.len()`. So it should be fine to use `len`. 

If we want robust memory measurements, I'm afraid we'll have to explore different approaches like arena allocators.

Reply:
> Joris has since changed the corresponding check on metric names to use .len()

@jan-auer are you referring to this line? The reason why it uses `.len()` is because metric names are now represented by `Arc<str>`, not `String` (see https://github.com/getsentry/relay/pull/3279).

https://github.com/getsentry/relay/blob/e614e9127c6909bd0a4acca464705dcebdc101e4/relay-metrics/src/aggregator.rs#L84

No heavy objections against this change, but the cleanest solution would be to have two different estimation functions for memory footprint and serialization cost.